### PR TITLE
Startup failure

### PR DESCRIPTION
**Pull request by terovirtanen**
### Closes :no_good_woman:
**[VEN-1525](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1525): berth federation gateway startup failure**

### CODE REVIEW

**Comment by terovirtanen** [comment]
Should cache set ? https://www.apollographql.com/docs/apollo-server/performance/cache-backends/

Gateway 'serviceList' migration to 'IntrospectAndCompose' will be done in different PR

pod log:
```
Open city profile GrahpQL API is available!
Berth reservations GrahpQL API is available!

> berth-federation@1.0.0 serve /app
> node --max-http-header-size=16000 dist/index.js

Persisted queries are enabled and are using an unbounded cache. Your server is vulnerable to denial of service attacks via memory exhaustion. Set `cache: "bounded"` or `persistedQueries: false` in your ApolloServer constructor, or see https://go.apollo.dev/s/cache-backends for other alternatives.
The `serviceList` option is deprecated and will be removed in a future version of `@apollo/gateway`. Please migrate to its replacement `IntrospectAndCompose`. More information on `IntrospectAndCompose` can be found in the documentation.
Server startup success.
🚀 Server ready at http://localhost:3000
```

File: src/dataSources.ts [hunk 1]
```diff
@@ -110,7 +111,7 @@ export class FileUploadDataSource extends RemoteGraphQLDataSource {
 
       return response;
     } catch (error) {
-      this.didEncounterError(error, httpRequest);
+//      this.didEncounterError(error, fetchRequest, httpResponse, context);
```
Comment (tuomas777) [inline src/dataSources.ts:114]: Is this commented out intentionally? If so, it would be nice if it had some comment on why that has been done 🙂

Comment (terovirtanen) [reply src/dataSources.ts]: My mistake, 'didEncounterError' has changed and I didn't do the change properly.
Anyway [didEncounterError](https://github.com/apollographql/federation/blob/f958fb964f9baa53edd149bcdf0d1217e7c39f0c/gateway-js/src/datasources/RemoteGraphQLDataSource.ts#L272) just throw error

**Review by nikomakela — APPROVED** [review]
code looks good to me. Tried the build locally also and it seems to work. Could be tested in staging server since it's a huge work to get the local environment up and running.

**Pull request merged**